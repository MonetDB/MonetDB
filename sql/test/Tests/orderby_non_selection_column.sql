-- some dumb table
create table lim_prob(dir varchar(30), test varchar(30), dir_test varchar(60));
-- some stupid values
insert into lim_prob values ('mydir1/', 'mytest1', 'mydir1/mytest1');  
insert into lim_prob values ('mydir2/', 'mytest3', 'mydir2/mytest3');  
insert into lim_prob values ('mydir1/', 'mytest2', 'mydir1/mytest2');  
insert into lim_prob values ('mydir1/', 'mytest4', 'mydir1/mytest4');  
insert into lim_prob values ('mydir2/', 'mytest1', 'mydir2/mytest1');  
insert into lim_prob values ('mydir2/', 'mytest2', 'mydir2/mytest2');  
insert into lim_prob values ('mydir1/', 'mytest3', 'mydir1/mytest3'); 
-- and two simple queries :D
select test     from lim_prob order by dir_test limit 10;
select dir      from lim_prob order by dir_test limit 10;
select dir_test from lim_prob order by dir,test limit 10;
-- clean up
drop table lim_prob;
