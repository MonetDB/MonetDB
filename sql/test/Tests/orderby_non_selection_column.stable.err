stderr of test 'orderby_non_selection_column` in directory 'sql/test` itself:


# 09:21:42 >  
# 09:21:42 >  Mtimeout -timeout 180 Mserver "--config=/ufs/niels/scratch/monetdb/Linux-x86_64/etc/MonetDB.conf" --debug=10 --set "monet_mod_path=/ufs/niels/scratch/monetdb/Linux-x86_64/lib/MonetDB4:/ufs/niels/scratch/monetdb/Linux-x86_64/lib/MonetDB4/lib:/ufs/niels/scratch/monetdb/Linux-x86_64/lib/MonetDB4/bin" --set "gdk_dbfarm=/ufs/niels/scratch/monetdb/Linux-x86_64/var/MonetDB4/dbfarm" --set "sql_logdir=/ufs/niels/scratch/monetdb/Linux-x86_64/var/MonetDB4/sql_logs" --set "xquery_logdir=/ufs/niels/scratch/monetdb/Linux-x86_64/var/MonetDB4/xquery_logs" --set mapi_port=33976 --set xrpc_port=48121 --set monet_prompt= --trace "--dbname=mTests_src_test" --dbinit=" module(sql_server);" ; echo ; echo Over..
# 09:21:42 >  


# 09:21:42 >  
# 09:21:42 >  Mtimeout -timeout 60 MapiClient -lsql -umonetdb -Pmonetdb --host=localhost --port=33976 
# 09:21:42 >  


# 09:21:43 >  
# 09:21:43 >  Done.
# 09:21:43 >  

