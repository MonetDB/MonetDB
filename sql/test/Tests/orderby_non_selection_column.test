statement ok
create table lim_prob(dir varchar(30), test varchar(30), dir_test varchar(60))

statement ok
insert into lim_prob values ('mydir1/', 'mytest1', 'mydir1/mytest1')

statement ok
insert into lim_prob values ('mydir2/', 'mytest3', 'mydir2/mytest3')

statement ok
insert into lim_prob values ('mydir1/', 'mytest2', 'mydir1/mytest2')

statement ok
insert into lim_prob values ('mydir1/', 'mytest4', 'mydir1/mytest4')

statement ok
insert into lim_prob values ('mydir2/', 'mytest1', 'mydir2/mytest1')

statement ok
insert into lim_prob values ('mydir2/', 'mytest2', 'mydir2/mytest2')

statement ok
insert into lim_prob values ('mydir1/', 'mytest3', 'mydir1/mytest3')

query T nosort
select test     from lim_prob order by dir_test limit 10
----
mytest1
mytest2
mytest3
mytest4
mytest1
mytest2
mytest3

query T nosort
select dir      from lim_prob order by dir_test limit 10
----
mydir1/
mydir1/
mydir1/
mydir1/
mydir2/
mydir2/
mydir2/

query T nosort
select dir_test from lim_prob order by dir,test limit 10
----
mydir1/mytest1
mydir1/mytest2
mydir1/mytest3
mydir1/mytest4
mydir2/mytest1
mydir2/mytest2
mydir2/mytest3

statement ok
drop table lim_prob

