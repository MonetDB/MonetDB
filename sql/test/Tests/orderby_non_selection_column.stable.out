stdout of test 'orderby_non_selection_column` in directory 'sql/test` itself:


# 18:28:19 >  
# 18:28:19 >   mserver5 "--config=/ufs/niels/scratch/monetdb/Linux-x86_64/etc/monetdb5.conf" --debug=10 --set gdk_nr_threads=4 --set "monet_mod_path=/ufs/niels/scratch/monetdb/Linux-x86_64/lib/MonetDB5:/ufs/niels/scratch/monetdb/Linux-x86_64/lib/MonetDB5/lib:/ufs/niels/scratch/monetdb/Linux-x86_64/lib/MonetDB5/bin" --set "gdk_dbfarm=/ufs/niels/scratch/monetdb/Linux-x86_64/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/niels/scratch/monetdb/Linux-x86_64/var/MonetDB5/sql_logs"  --set mapi_open=true --set xrpc_open=true --set mapi_port=39921 --set xrpc_port=45986 --set monet_prompt= --trace  "--dbname=mTests_src_test" --set mal_listing=0 "--dbinit= include sql;" ; echo ; echo Over..
# 18:28:19 >  

# MonetDB server v5.9.0, based on kernel v1.27.0
# Serving database 'mTests_src_test', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008- MonetDB B.V., all rights reserved
# Visit http://monetdb.cwi.nl/ for further information
# Listening for connection requests on mapi:monetdb://alf.ins.cwi.nl:39921/
# MonetDB/SQL module v2.27.0 loaded

# falling back to direct mode 
# falling back to direct mode 
# falling back to direct mode 
# falling back to direct mode 
#function user.main():void;
#    clients.quit();
#end main;


# 09:21:42 >  
# 09:21:42 >  Mtimeout -timeout 60 MapiClient -lsql -umonetdb -Pmonetdb --host=localhost --port=33976 
# 09:21:42 >  

[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
% sys.lim_prob # table_name
% test # name
% varchar # type
% 7 # length
[ "mytest1"	]
[ "mytest2"	]
[ "mytest3"	]
[ "mytest4"	]
[ "mytest1"	]
[ "mytest2"	]
[ "mytest3"	]
#select dir      from lim_prob order by dir_test limit 10;
% sys.lim_prob # table_name
% dir # name
% varchar # type
% 7 # length
[ "mydir1/"	]
[ "mydir1/"	]
[ "mydir1/"	]
[ "mydir1/"	]
[ "mydir2/"	]
[ "mydir2/"	]
[ "mydir2/"	]
#select dir_test from lim_prob order by dir,test limit 10;
% sys.lim_prob # table_name
% dir_test # name
% varchar # type
% 14 # length
[ "mydir1/mytest1"	]
[ "mydir1/mytest2"	]
[ "mydir1/mytest3"	]
[ "mydir1/mytest4"	]
[ "mydir2/mytest1"	]
[ "mydir2/mytest2"	]
[ "mydir2/mytest3"	]
#drop table lim_prob;

# 17:46:36 >  
# 17:46:36 >  Done.
# 17:46:36 >  

